webapp/src/lib/Ui.ts

```diff
@@ -47,6 +49,33 @@ export async function showDialogInfo(dialog: MatDialog, title: string, text: str
     });
 }
 
+export async function storeCredential(dialog: MatDialog, title: string,
+                                      store: () => Promise<void>) {
+    return showTransactions(dialog, title, async (progress: TProgress) => {
+        progress(50, "Storing Credential");
+        await store();
+    });
+}
+
+export type TProgress = (percentage: number, text: string) => void;
+export type TWorker = (progress: TProgress) => Promise<void>;
+
+export async function showTransactions(dialog: MatDialog, title: string, worker: TWorker): Promise<string> {
```

Comment:
Why is it returning the result of the dialog? There is not input by the user, correct? Returning the potentially long-to-wait result would be more useful IMO.

Reply:
As the dialog catches the error of the callback, I return the error here, if there is one. This looked better, as the dialog itself already shows the error to the user, so I didn't want to `throw` it again.

Reply:
Ho, right, for the error case, we don't need to throw it also, but for the happy path, we might want to retrieve the result.
To better understand why I'm thinking of using it this way: in `admin/contacts/contacts.components.ts:ContactsComponent.darcInstanceAdd`, instead of wrapping it all in callback style, awaiting on some result of the callback from `storeCredential` allows to separate the logic of "letting the user choose a darc instance from a list" from the "where to store the actions".